### PR TITLE
Refactor products model

### PR DESCRIPTION
**Pull request by kapoliub**

### CODE REVIEW

**Review by Iryna-Bzdyr — COMMENTED** [review]
You also have to do refactor product.service.js.
As now we use object id reference.
So that, for example when you want get product by id, you have to use populate method
You code will be look like this:
 getProductById(id) {
    return Product.findById(id).populate(
      'pattern',
    ).populate(
      'colors',
    ).populate(
      'closures',
    );
  }
Please read about populate method
https://qna.habr.com/q/398941

**Review by ahd3r — APPROVED** [review]

File: modules/product/product.service.js [hunk 1]
```diff
@@ -17,11 +17,19 @@ const { uploadProductImages } = require('./product.utils');
 
 class ProductsService {
```
Comment (moran711) [inline modules/product/product.service.js:18]: You need to populate all collections in all methods.

File: modules/product/product.model.js [hunk 2]
```diff
@@ -17,11 +16,20 @@ const productSchema = new mongoose.Schema({
...
+    {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: 'Colors',
+    },
+  ],
+  pattern: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Pattern',
+  },
+  closure: {
```
[the remark this answers is not in the document]
Comment (kapoliub) [reply modules/product/product.model.js]: fixed

**Review by Iryna-Bzdyr — APPROVED** [review]

File: modules/product/product.service.js (hunk 1 above)
Comment (kapoliub) [reply modules/product/product.service.js:18]: fixed

**Review by moran711 — APPROVED** [review]

**Pull request merged**